src/core/components/courses/ProfileOrganization.vue

```diff
@@ -534,15 +530,20 @@ export default {
     async updateSalesRepresentative () {
       try {
         this.interlocutorModalLoading = true;
-        this.v$.tempInterlocutorId.$touch();
-        if (this.v$.tempInterlocutorId.$error) return NotifyWarning('Champ(s) invalide(s)');
+        this.v$.tempInterlocutor.$touch();
+        if (this.v$.tempInterlocutor.$error) return NotifyWarning('Champ(s) invalide(s)');
 
         const payload = {
-          salesRepresentative: this.tempInterlocutorId,
-          ...(get(this.course, 'contact._id') === get(this.course, 'salesRepresentative._id') &&
-            { contact: this.tempInterlocutorId }
+          salesRepresentative: this.tempInterlocutor._id,
+          ...(
+            (
+              this.tempInterlocutor.isContact ||
+              get(this.course, 'contact._id') === this.course.salesRepresentative._id
+            ) &&
+            { contact: this.tempInterlocutor.isContact ? this.tempInterlocutor._id : '' }
           ),
```

Comment:
Que penses-tu d'envoyer le champs a chaque fois ? Ce n'est pas toujours utile mais j'ai l'impression que ca ne serait pas tres genant et que ce serait plus lisible 🤔

Reply:
ça complique encore plus le code non ? il faut rajouter une 3e condition où si le contact est un autre interlocuteur il faut envoyer cet autre interlocuteur comme contact

Reply:
On ne pourrais pas dans tous les cas mettre le champs `{ contact: this.tempInterlocutor.isContact ? this.tempInterlocutor._id : '' }` ?